pkg/plugin/plugin.go

```diff
@@ -296,10 +391,79 @@ func (g *GlooMatchedRouteTable) matchRoutes(logCtx *logrus.Entry, rollout *v1alp
 					CanaryOrPreviewDestination: canary,
 				},
 			}
-			logCtx.Debugf("adding destination %+v", dest)
 			g.HttpRoutes = append(g.HttpRoutes, dest)
 		}
 	} // end range httpRoutes
 
 	return nil
 }
+
+func buildGlooHTTPRoute(name string, matcher *solov2.HTTPRequestMatcher, canaryDestination *SimpleDestinationReference) *GlooMatchedHttpRoutes {
+	return &GlooMatchedHttpRoutes{
+		HttpRoute: &networkv2.HTTPRoute{
+			Name: name,
+			Matchers: []*solov2.HTTPRequestMatcher{
+				matcher,
+			},
+			ActionType: &networkv2.HTTPRoute_ForwardTo{
+				ForwardTo: &networkv2.ForwardToAction{
+					Destinations: []*solov2.DestinationReference{
+						{
+							RefKind: &solov2.DestinationReference_Ref{
+								Ref: &solov2.ObjectReference{
+									Name:      canaryDestination.Reference.Name,
+									Namespace: canaryDestination.Reference.Namespace,
+								},
+							},
+							Port: &solov2.PortSelector{
+								Specifier: &solov2.PortSelector_Number{
+									Number: canaryDestination.Port.Number,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildGlooMatches(headerRouting *v1alpha1.SetHeaderRoute) *solov2.HTTPRequestMatcher {
+	matcher := &solov2.HTTPRequestMatcher{
+		Name:    headerRouting.Name + "-matcher",
+		Headers: []*solov2.HeaderMatcher{},
+	}
+
+	for _, m := range headerRouting.Match {
+		var isRegex bool
+		var matchValue string
+		if m.HeaderValue.Exact != "" {
+			matchValue = m.HeaderValue.Exact
+		} else if m.HeaderValue.Regex != "" {
+			matchValue = m.HeaderValue.Regex
+			isRegex = true
+		} else if m.HeaderValue.Prefix != "" {
+			// error, do nothing?
+			// try to convert to a regex for the user?
+			panic("TODO: impl some handling of prefix")
```

Comment:
I think[ strings.HasPrefix](https://pkg.go.dev/strings#HasPrefix) would satisfy this, right?

Reply:
The issue is we need to either panic, error out or map the Rollout prefix field onto the solo api somehow. Near as I can tell we have no real support for this directly (I may be mistaken). Probably for this release we should just document that Prefix is not supported and users are probably fine with this since they're already used to that limitation in solo API?

Reply:
FWIW, the panic doesn't actually bring down argo-rollouts its just very loud about being unhappy

Reply:
```golang
		} else if m.HeaderValue.Prefix != "" {
			matchValue = "^" + regexp.QuoteMeta(m.HeaderValue.Prefix)
			isRegex = true
		}
```

This is another option. Seems ok actually

Reply:
produces this in the route:
```yaml
    matchers:
    - headers:
      - name: version
        regex: true
        value: ^canary

```

Reply:
Nice, that works very well I think. It's less about the runtime of the plugin and more about what kind if support we're giving as support to the user. Prefix is a pretty common use case.

Reply:
Resolved! (Idk why, Mark As Resolved isn't showing up for me)